public/blog/text2sql.md

```diff
@@ -0,0 +1,87 @@
+
+![gpt](/blog/assets/text2sql/gpt.webp)
+
+These tools started out to bridge the gap between non-tech users and databases, by allowing them to interact with databases using natural language and reduce the barrier to accessing and analyzing data. But with the advance of AI models, these tools now support more advanced features such as handling complex queries, joining multiple tables, or even supporting natural language conversations.
+
+They can also help improve productivity by automating the process of generating SQL queries, thereby saving time and effort.
+
+In this edition of Star History monthly, we have compiled a collection of open-source Text2SQL tools.
+
+- [Star History Monthly | Open-source Text2SQL Tools](#star-history-monthly--open-source-text2sql-tools)
+  - [Chat2DB](#chat2db)
+  - [SQL Chat](#sql-chat)
+  - [Vanna](#vanna)
+  - [DuckDB-NSQL](#duckdb-nsql)
+  - [Langchain](#langchain)
+  - [Awesome Text2SQL](#awesome-text2sql)
+  - [To Wrap up](#to-wrap-up)
+  - [Lastly](#lastly)
+
+## Chat2DB
+
+[Chat2DB](https://github.com/chat2db/chat2db) aims to be a general-purpose SQL client and reporting tool that incorporates AI capabilities from the start. It supports connection to a handful of databases including MySQL, Postgres, Oracle, SQL Server, SQLite, ClickHouse and more.
+
+![chat2db](/blog/assets/text2sql/chat2db.webp)
+
+There was a bit of [drama](https://www.linkedin.com/feed/update/urn:li:activity:7153040091454611456/) involving Chat2DB a while ago, we won't get into details here but curious to know what you think.
+
+## SQL Chat
+
+[SQL Chat](https://github.com/sqlchat/sqlchat) is a chat-based SQL client, and you can use natural language to communicate with your database to implement operations, such as query, modification, addition, and deletion (!) of the database.
+
+It currently supports MySQL, Postgres, SQL Server and TiDB serverless.
+
+![sqlchat](/blog/assets/text2sql/sqlchat.webp)
+
+It's [open-sourced by Bytebase](https://www.bytebase.com/blog/sql-chat/), a database migration tool for teams.
+
+## Vanna
+
+[Vanna](https://github.com/vanna-ai/vanna) is a Python library that allows the training of an RAG model with queries, DDL, and documentation from a database.
+
+![vanna](/blog/assets/text2sql/vanna.webp)
+
+You can use Vanna as is, or build your own custom UI with an existing tool (e.g. Streamlit, Slack).
+
+It was open-sourced July, 2023, but [got really popular](https://star-history.com/#vanna-ai/vanna&Date) this January.
+
+## DuckDB-NSQL
+
+[DuckDB-NSQL](https://github.com/NumbersStationAI/DuckDB-NSQL) is a Text2SQL LLM built for local DuckDB SQL analytics tasks, built by DuckDB and Numbers Station. This can certainly help users leverage the full power of DuckDB and its analytic potential, without having to go forth-and-back between the DuckDB documentation and the SQL shell.
```

Comment:
built by MonthDuck and Number Stations. And please include the backlink to both of them.

Reply:
back-and-force